### ISSUE TITLE
Application should not be editable after creation

### ISSUE THREAD

**Issue by jaredlockhart** [mozilla/experimenter#7083]
I just noticed a test was updating application for a created experiment but that shouldn't be possible since that can invalidate other fields.  Application should only be saveable on creation and our logic should prevent it from being changed, even in tests.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-2215)

**Comment by cherish2003** [comment]
Hello @yashikakhurana. I am an Outreachy applicant and I would love to participate in this experimenter project by working on this issue.

**Comment by yashikakhurana** [comment]
@cherish2003 wanted to confirm again since other issue is assigned, once you finish that, I can assign it to you.

**Comment by cherish2003** [comment]
@yashikakhurana okay mam

**Comment by prernadabi23** [comment]
Hello, Can I take on this issue?
@yashikakhurana

**Comment by yashikakhurana** [comment]
@prernadabi23 since I know you are having issue with your laptop, I am removing the assignment for now so it becomes available to be pick up by other applicant, meanwhile you can always tag me on the issue if its still not taken , thank you :)

**Comment by suhana832** [comment]
@yashikakhurana can I work on this task ?

**Comment by cherish2003** [comment]
@yashikakhurana i want to work on this issue again can you assign me

**Comment by suhana832** [comment]
> I just noticed a test was updating application for a created experiment but that shouldn't be possible since that can invalidate other fields. Application should only be saveable on creation and our logic should prevent it from being changed, even in tests.
> 
> ┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-2215)

can you provide screenshot or more details

**Comment by suhana832** [comment]
@jaredlockhart can you please elaborate the issue more ?

**Comment by yashikakhurana** [comment]
@suhana832 As you will see in some of the test cases, python test cases- when we create an experiment, we pass the application too, but you will notice that in the same test case if you will try to update the application of the experiment, it will allow you to do so, but in actual it shouldn't let you modify application, test should fail, so we have to change our logic in the Django code, please let me know if you have any more questions

**Comment by suhana832** [comment]
i cant work on this issue : (

**Comment by cherish2003** [comment]
@yashikakhurana can you specify the test file unable to locate it

**Comment by yashikakhurana** [comment]
like in here-https://github.com/mozilla/experimenter/blob/b4824f9ea47b2d4f57cb480cf1acdf22e19e1a64/experimenter/experimenter/experiments/tests/api/v5/test_serializers/test_nimbus_experiment_serializer.py#L270, the idea is add a new test which only changes the application, it will pass now, but once you add a check/logic to not to allow it, your test will fail, which we exactly wants, and then you can update your newly added test and all the other test which should fail

**Comment by KramStyles** [comment]
@yashikakhurana I would love to work on this. 
If I understand correctly, I should add a test to assert if the application can be updated after it's creation (which should not be allowed). Then I should update the logic in the main code to stop the application from been updated after been created?

**Comment by yashikakhurana** [comment]
@KramStyles that's correct

**Comment by KramStyles** [comment]
Hello @yashikakhurana ,

I'm sorry I've not made a push for this task. I've already started working on it but my PC crashed.

My docker really slowed down my system as it takes more than 10 minutes to start the application and 2 minutes for a reload.  I read it's because docker runs over the WSL layer which can heavily affect its speed.

 I decided to dual boot Linux and Windows so I can use Linux for the Experimenter project but in the process everything crashed. 

Currently I'm trying to fix my laptop

**Comment by nox1134** [comment]
Hi @yashikakhurana! If no one is working on this issue, I'd like to work on it.
Could you please assign the issue to me?

**Comment by yashikakhurana** [comment]
@nox1134 absolutely, thank you for your interest, let me know if you need any help 🙏
